examples/chat/tui/commands.nim

```diff
@@ -27,6 +27,62 @@ const hashCharSet = {'#'}
 # `command` proc should implement the appropriate logic to deal with those
 # values
 
+# AddCustomToken ----------------------------------------------------------------
+
+proc help*(T: type AddCustomToken): HelpText =
+  let command = "addcustomtoken"
+  HelpText(command: command, aliases: aliased[command], parameters: @[
+    CommandParameter(name: "address", description: "Address of the " &
+      "custom token."),
+    CommandParameter(name: "name", description: "Name of " &
+      "the custom token."),
+    CommandParameter(name: "symbol", description: "Symbol of the " &
+      "custom token."),
+    CommandParameter(name: "color", description: "(Optional) Color (in hex) for " &
+      "custom token."),
+    CommandParameter(name: "decimals", description: "(Optional) Number of decimals to use for " &
+      "the custom token.")
+    ], description: "Creates a new custom token ")
+
+proc new*(T: type AddCustomToken, args: varargs[string]): T =
+  var
+    address = ""
+    name = ""
+    symbol = ""
+    color = ""
+    decimals = ""
+  if args.len > 2:
+    address = args[0]
+    name = args[1]
+    symbol = args[2]
+  if args.len > 4:
+    color = args[3]
+    decimals = args[4]
+
+  T(address: address, name: name, symbol: symbol, color: color, decimals: decimals)
+
+proc split*(T: type AddCustomToken, argsRaw: string): seq[string] =
+  argsRaw.split(" ")
+
+proc command*(self: ChatTUI, command: AddCustomToken) {.async, gcsafe,
+  nimcall.} =
+
+  try:
+    if command.address == "":
+      self.wprintFormatError(epochTime().int64,
+        "address cannot be empty, please provide an address.")
+    elif command.name == "":
+      self.wprintFormatError(epochTime().int64,
+        "name cannot be empty, please provide a name.")
+    elif command.symbol == "":
+      self.wprintFormatError(epochTime().int64,
+        "symbol cannot be empty, please provide a symbol.")
+    else:
+      asyncSpawn self.client.addCustomToken(command.address, command.name, command.symbol, command.color, command.decimals)
+  except:
+    self.wprintFormatError(epochTime().int64, "invalid arguments.")
```

Comment:
Will the source of error always be invalid arguments? Maybe we could catch the exception and display the error message, like so:
```suggestion
  except Exception as e:
    self.wprintFormatError(epochTime().int64, "Error adding custom token: " & e.msg)
```

Reply:
Thanks, fixed!